datasource/src/test/scala/quasar/physical/mongo/interpreter/MaskSpec.scala

```diff
@@ -18,84 +18,80 @@ package quasar.physical.mongo.interpreter
 
 import slamdata.Predef._
 
+import org.bson._
 import org.specs2.mutable.Specification
 
 import quasar.api.table.ColumnType
 import quasar.common.CPath
-import quasar.physical.mongo.MongoExpression
-import quasar.physical.mongo.{Version, Aggregator, MongoExpression => E}, E.helpers._
+import quasar.physical.mongo.Version
+import quasar.physical.mongo.expression._
 
 class MaskSpec extends Specification {
-  "min versions" >> {
-    "older than $type" >> {
-      Mask("root", Version.zero, Map(CPath.parse(".") -> ColumnType.Top)) must beNone
-    }
-    "newer than $type" >> {
-      Mask("root", Version.$type, Map(CPath.parse(".") -> ColumnType.Top)) must beSome
-    }
-  }
-
   "examples" >> {
-    def evalMask(masks: Map[CPath, Set[ColumnType]]): Option[List[Aggregator]] =
-      Mask("root", Version.$type, masks)
+    def evalMask(masks: Map[CPath, Set[ColumnType]]): Option[List[Pipe]] =  Mask("root", masks)
 
-    val rootKey = E.key("root")
+    val rootKey = O.key("root")
 
-    val undefined = E.key("root_non_existent_field")
+    val undefined = O.key("root_non_existent_field")
 
-    def mkExpected(e: MongoExpression) =
-      Some(List(Aggregator.project(e), Aggregator.notNull("root")))
+    def mkExpected(e: Map[String, Expr]) =
+      Some(List(Pipeline.$project(e), Pipeline.NotNull("root")))
 
-    def typeEq(prj: MongoExpression, s: String): MongoExpression =
-      equal(typeExpr(prj), E.String(s))
+    def typeEq(prj: Expr, s: String): Expr =
+      O.$eq(List(O.$type(prj), O.string(s)))
 
-    def isObjectFilter(x: MongoExpression, y: MongoExpression): MongoExpression =
-      cond(typeEq(x, "object"), y, undefined)
+    def isObjectFilter(x: Expr, y: Expr): Expr =
+      O.$cond(typeEq(x, "object"), y, undefined)
 
+    def pipeEqual(a: Option[List[Pipe]], b: Option[List[Pipe]]) = {
+      val compile: List[Pipe] => Option[List[BsonDocument]] = x => compilePipeline(Version.$objectToArray, x)
+      val aExpr = a flatMap compile
```

Comment:
`$or` and `$let` optimization happens on `CoreOp`. It's inconvenient too 🙂

Reply:
I mean, both the expectation and the pipe under test are compiled in the same way. If I change `compilePipeline` to return simply `None` then the tests will still be green.

It would be better if the expectation wouldn't use the prod code